add-patch.c

```diff
@@ -1184,8 +1184,10 @@ static int read_single_character(struct add_p_state *s)
 {
 	if (s->s.use_single_key) {
 		int res = read_key_without_echo(&s->answer);
-		printf("%s\n", res == EOF ? "" : s->answer.buf);
-		return res;
+		if (res != EOF) {
+			printf("%s\n", s->answer.buf);
```

Comment:
This is a change of behavior when res == EOF - is that intended? If so maybe call it out in the commit message